components/sync15/README.md

```diff
@@ -5,4 +5,108 @@ data stores that want to sync against a Firefox Sync v1.5 sync server.
+The important part is that the API is conceptually quite simple - there are
+arbitrary "collections", indexed by a GUID, and lacking traditonal database
+concepts like joins. Because the payload is encrypted, there's very little
+scope for it to be much smarter. Thus it's reasonably easy to create a fairly
+generic abstraction over the API that can be easily reused.
+
+Back in the deep past, we found ourselves with 2 different components that
+needed to sync against a sync v1.5 server. The apps using these components
+didn't have schedulers or any UI for choosing what to sync - so these
+components just looked at the existing state of the engines on the server and
+synced if they were enabled.
+
+This was also pre-megazord - the idea was that apps could choose from a "menu"
+of components to include - so we didn't really want to bind these components
+together. Therefore, there was no concept of "sync all" - instead, each of the
+components had to be synced individually. So this component stated out as more
+of a "library" than a "component" which individual components could reuse - and
+each of these components was a "syncable store".
+
+Fast forward to Fenix and we needed a UI for managing all the engines supported
+there, and a single "sync now" experience etc - so we also have a sync_manager
+component - [see its README for more](../components/sync_manager/README.md).
+But even though it exists, there are still some parts of this component that
+reflect these early days - for example, it's still possible to sync just a
+single component using sync15 (ie, without going via the "sync manager"),
+although this isn't used and should be removed - the "sync manager" allows you
+to choose which engines to sync, so that should be used exclusively.
+
+## Metadata
+
+There's some metadata associated with a sync. Some of the metadata is "global"
+to the app (eg, the enabled state of engines, information about what servers to
+use, etc) and some is specific to an engine/store (eg, timestamp of the
+server's collection for this engine/store, guids for the collections, etc).
+
+We made the decision early on that no storage should be done by this
+component:
+
+* The "global" metadata should be stored by the application - but because it
+  doesn't need to interpret the data, we do this with an opaque string (that
+  is JSON, but the app should never assume or introspect that)
+
+* Each store/engine should store its own metadata, so we don't end up in the
+  situation where, say, a database is moved between profiles causing the
+  metadata to refer to a completely different data set. So each store/engine
+  stores its metadata in the same database as the data itself, so if the
+  database is moved or copied, the metadata comes with it)
+
+## Sync Implementation
+
+The core implementation does all of the interaction with things like the
+token-server, the `meta/global` and `info/collections` collections, etc. It
```

Comment:
I'm not sure how important understanding the details of these interactions are, but is it worthwhile to include links to further info on `meta/global` or `info/collections`?

Reply:
I would if I could find such links :/ The doc I linked to above is the closest I can find, but they are fairly vague on that.